packages/ocs/hooks/useStorageClassQueryFilter.ts

```diff
@@ -0,0 +1,38 @@
+import { StorageClassModel } from '@odf/shared/models';
+import { getName } from '@odf/shared/selectors';
+import { StorageClassResourceKind } from '@odf/shared/types';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { getSpecificClusterCephSC } from '../utils';
+
+const scResource: WatchK8sResource = {
+  kind: StorageClassModel.kind,
+  isList: true,
+};
+
+export const useStorageClassQueryFilter = (
+  clusterNs: string
+): [string, boolean, unknown] => {
+  const [scData, scDataLoaded, scDataError] =
+    useK8sWatchResource<StorageClassResourceKind[]>(scResource);
+
+  let filteredStorageClasses: StorageClassResourceKind[] | string[] = [];
+
+  if (!clusterNs) {
+    /**
+     * This is for ODF Client Mode use case only ("clusterNs" will be "undefined" when this hook will be called from client-console).
+     * ODF client mode doesn't deploy ODF StorageCluster hence no "clusterNs", moreover Route for dashboards will be differnt as well.
+     * For scenarios where Client & Provider are on same OCP, client-console will be disabled (as per design).
+     * Considering all the StorageClasses in the cluster in such case.
+     */
+    filteredStorageClasses = ['.*'];
```

Comment:
```suggestion
    filteredStorageClasses = ['.*'];
    return [filteredStorageClasses, true, null]
```
No need to process sc data incase of client mode.